lua/starfall/libs_sh/holograms.lua

```diff
@@ -332,45 +356,68 @@ else
 		end
 	end
 
+	local function clearParentFix(ent)
+		if ent.sf_parent then
+			ent.sf_parent.sf_children[ent] = nil
+			ent.sf_parent = nil
+		end
+	end
+	
+	local function setParentFix(ent, parent, bone, func)
+		if not parent.sf_children then
+			parent.sf_children = {}
+		end
+		
+		if ent.sf_parent and ent.sf_parent ~= parent then
+			ent.sf_parent.sf_children[ent] = nil
+		end
+		
+		ent.sf_parent = parent
+		parent.sf_children[ent] = {
+			func = func,
+			bone = bone, -- or attachment, dep. on `func`
+			pos  = parent:WorldToLocal(ent:GetPos()),
+			ang  = parent:WorldToLocalAngles(ent:GetAngles())
+		}
+	end
+	
 	--- Parents a hologram
-	-- @param Entity? ent Entity parent (nil to unparent)
+	-- @param Entity? parent Entity parent (nil to unparent)
 	-- @param number? attachment Optional attachment ID
-	function hologram_methods:setParent(ent, attachment)
-
+	function hologram_methods:setParent(parent, attachment)
 		local holo = getholo(self)
-
 		checkpermission(instance, holo, "hologram.setParent")
 
-		if ent ~= nil then
-			local parent = getent(ent)
-
-			if attachment == nil then attachment = -1 end
+		if parent ~= nil then
+			parent = getent(parent)
+			attachment = attachment or -1
 			checkluatype(attachment, TYPE_NUMBER)
-
-			if not parent.sf_children then
-				parent.sf_children = {}
-			end
-
-			if holo.sf_parent then
-				holo.sf_parent.sf_children[holo] = nil
-			end
-
-			parent.sf_children[holo] = {attachment, parent:WorldToLocal(holo:GetPos()), parent:WorldToLocalAngles(holo:GetAngles())}
-			holo.sf_parent = parent
-
+			
+			setParentFix(holo, parent, attachment, "SetParent")
 			holo:SetParent(parent, attachment)
-
 		else
-
-			if holo.sf_parent then
-				holo.sf_parent.sf_children[holo] = nil
-			end
-
-			holo.sf_parent = nil
+			clearParentFix(holo)
 			holo:SetParent()
-
 		end
-
+	end
+	
+	--- Parents a hologram to the specified parent's bone
+	-- @param Entity? parent Entity parent (nil to unparent)
+	-- @param number bone Bone ID
+	function hologram_methods:followBone(parent, bone)
+		local holo = getent(self)
+		checkpermission(instance, holo, "hologram.setParent")
+		
+		if parent ~= nil then
+			parent = getent(parent)
+			checkluatype(bone, TYPE_NUMBER)
+			
```

Comment:
parentChainTooLong should be checked in the clientside too right?

Reply:
Clientside Hologram didn't have the check originally and I couldn't find much info about this bug so I assumed it only affectes serverside parents.  
I can implement it though.

Reply:
Probably good to be safe